bash_scripting_101/bash_scripting_101.md

```diff
@@ -37,241 +39,443 @@ script: https://kit.fontawesome.com/83b2343bd4.js
+
+There are some text editors already installed in your bash shell, like nano. These are useful because they don't require new downloads, but can have a steep learning curve since you need to learn new commands to open, save, and quit in those programs.
+
+Other, more user-friendly, text editors exist and can be downloaded. [Atom](https://atom.io/) is a particularly good free text editor that includes helpful tools like color coding text.
+
+
+**`nano`**
+
+The `nano` text editor is already in Bash Shell. You can type `nano` to start a new file, or `nano file_name` to open a file that already exists. You will need to use keyboard commands to save, or WriteOut, and exit the program, but they are all helpfully listed at the bottom of your window when you are in nano.
+
+For example to exit nano, the `^X` reminder at the bottom means that you should hold down the `control` button at the same time as the key `X`. You will then be helpfully asked if you want to "Save modified buffer," or in other words, save your changes. Type `Y` and you will be asked for the "File name to write." If you are working in a new file, give it a name here! Otherwise you can press `enter`/`return` to accept the name the file already had.  
+
 
-## Glossary
+**Atom**
+
+Atom is a free text editor that you can download from [https://atom.io/](https://atom.io/). Once you open the Atom program, you will see "File" and "Edit" menus like you might expect from a word processing program. You can open files, save them, as well as create new folders and files from Atom's "File" menu.
+
+
+**Vim**
+
+Vim may be the default editor in your shell, but that doesn't mean it is the best one. It is notoriously un-friendly to new users so we don't recommend it as your first (or second) choice of text editor.
+
+<div class = "help">
+
+**Stuck in Vim?**
+
+The default text editor on macOS and Linux is usually set to Vim, which is not famous for being intuitive. That is why we don't recommend it as a text editor. If you accidentally find yourself stuck in it, hit the `Esc` key, followed by `:`+`Q`+`!` (colon, lower-case 'q', exclamation mark), then hitting Return to return to the shell.
+</div>
+
+
+### Quiz: Editing files
+
+1. You want to make a new folder inside `learning_bash` called `data` and put two files in it, `data_1` and `data_2`.
+
+```
+cd ~/learning_bash
+______A_______
+______B_______
+touch data_1 data_2
+```
+What code should go in lines A and B?
+
+Line A should be:
+
+[[ `touch data`| (`mkdir data`) | `cd data` ]]
+
+Line B should be:
+
+[[ `touch data`| `mkdir data` | (`cd data`) ]]
+***
+<div class = "answer">
+You must first make the directory `data` and then move into that directory with `cd data`.
+
+The command `touch data` will create a file named `data`, not a directory that can contain files.
+</div>
+***
+
+
+2. Now that you have created the files `data_1` you want to put some content into it. How would you make the file `data_1` contain the text `sample 1: 45`?
```

Comment:
This isn't working for me, I get the error "Unexpected identifier 'sample'. Expected ';' after variable declaration."

Reply:
@pm0kjp 
Are you getting this error in your CLI? or in entering the answer in Liascript? I haven't been able to reproduce it.

Reply:
I never got exactly the same error message you got, but I changed the whole format of the question to multiple choice because it seems having input that includes `"` and also allowing for multiple correct answers is every so slightly too hard.